pkg/resources/opnicluster/services.go

```diff
@@ -285,6 +287,73 @@ func (r *Reconciler) pretrainedModelDeployment(
 	}
 }
 
+func (r *Reconciler) workloadDrain() (resourceList []resources.Resource, retError error) {
+	resourceList = []resources.Resource{}
+	workloadDeployment, err := r.workloadDrainDeployment()
+	if err != nil {
+		panic(err)
+	}
+	resourceList = append(resourceList, workloadDeployment)
+	return resourceList, nil
+}
+
+func (r *Reconciler) workloadDrainDeployment() (resources.Resource, error) {
+	return func() (runtime.Object, reconciler.DesiredState, error) {
+		labels := resources.CombineLabels(
+			r.serviceLabels(v1beta2.DrainService),
+		)
+		imageSpec := r.serviceImageSpec(v1beta2.DrainService)
+		lg, _ := logr.FromContext(r.ctx)
+		lg.V(1).Info("generating workload DRAIN deployment")
+		envVars, volumeMounts, volumes := r.genericEnvAndVolumes()
+		s3EnvVars := r.s3EnvVars()
+		envVars = append(envVars, s3EnvVars...)
+		envVars = append(envVars,
+			corev1.EnvVar{
+				Name:  "FAIL_KEYWORDS",
```

Comment:
We should move some of these config items out of environment variables and into a config file

Reply:
https://github.com/rancher/opni/issues/756